packages/sil-governance/src/routes/show-electorate/index.tsx

```diff
@@ -30,14 +30,20 @@ const ShowElectorate = ({ match }: RouteComponentProps<Params>): JSX.Element =>
   const [electorate, setElectorate] = useState<Electorate>();
 
   useEffect(() => {
+    let isSubscribed = true;
+
     const updateElectorates = async (): Promise<void> => {
       // in DOM tests, governor is not set
-      if (governor) {
+      if (governor && isSubscribed) {
         const electorates = await governor.getElectorates();
         setElectorates(electorates);
       }
```

Comment:
And here. The idea is to avoid calling state change after all async operations. Because those async operations can be finished after component destruction.